[Delivers #93381468] Add an endpoint for feedback, along with its form

cmc333333 opened the pull request:
Also tweaks the logic around js filters to allow for logic when an option is _not_ selected. There are currently no links to this form.

## initial form
![screen shot 2015-06-11 at 5 39 57 pm](https://cloud.githubusercontent.com/assets/326918/8119459/9ab41bae-1063-11e5-8aea-f795c0ad305a.png)

## is a bug
![screen shot 2015-06-11 at 5 40 08 pm](https://cloud.githubusercontent.com/assets/326918/8119461/9ab7d3fc-1063-11e5-9b08-bf66507b4334.png)

## submitted (not logged in)
![screen shot 2015-06-11 at 5 40 29 pm](https://cloud.githubusercontent.com/assets/326918/8119458/9ab3c6d6-1063-11e5-8fe8-4db8eb23474c.png)

## submitted (logged in)
![screen shot 2015-06-11 at 5 41 00 pm](https://cloud.githubusercontent.com/assets/326918/8119460/9ab4ab00-1063-11e5-9417-c4ef3961a3a7.png)

## email
![screen shot 2015-06-11 at 5 41 37 pm](https://cloud.githubusercontent.com/assets/326918/8119457/9ab34030-1063-11e5-921e-a33c008b7393.png)

afeld commented on app/controllers/feedback_controller.rb:
```diff
@@ -0,0 +1,19 @@
+        message += "\nuser: #{current_user.email_address}"
+      end
+      CommunicartMailer.feedback(message).deliver
+    end
```
We might want to move this logic to a Plain Old Ruby Object down the road.

cmc333333 replied:
I'll move some of the logic into the mailer as a compromise

afeld commented on app/mailers/communicart_mailer.rb:
```diff
@@ -66,6 +66,18 @@ def comment_added_email(comment, to_email)
+      subject: 'Feedback submission',
+      from: default_sender_email,
+      body: message
+    )
```
How about adding them as the `reply_to`? Just to save us a step.

cmc333333 replied:
Good idea

afeld commented:
Looks good! Kind of bummed that we need to implement/maintain this ourselves and can't use something off-the-shelf, though :confused:

cmc333333 commented:
Updated to simplify the logic. Added the user as a CC (so that they will receive a receipt)

cmc333333 commented:
Updated to link to the feedback form on standard pages, the landing page, and 400s. The 500 template includes a mailto link instead. Added a little logic so that the feedback link does not appear on the feedback form.

Looks like:
### Landing Page
![screen shot 2015-06-15 at 8 40 06 pm](https://cloud.githubusercontent.com/assets/326918/8173684/490a34ca-139f-11e5-93c5-afe2bfc09f28.png)
### Normal Pages
![screen shot 2015-06-15 at 8 40 22 pm](https://cloud.githubusercontent.com/assets/326918/8173683/4909d9a8-139f-11e5-98a3-9a6d318d3d83.png)
### 404s
![screen shot 2015-06-15 at 8 39 45 pm](https://cloud.githubusercontent.com/assets/326918/8173685/490df510-139f-11e5-9fc6-cb63311e203a.png)
### 500
![screen shot 2015-06-15 at 8 22 50 pm](https://cloud.githubusercontent.com/assets/326918/8173687/4911f9ee-139f-11e5-946d-d0916d88b6b3.png)
### 500 email
![screen shot 2015-06-15 at 8 22 56 pm](https://cloud.githubusercontent.com/assets/326918/8173686/490feb54-139f-11e5-9f8f-edbad61d8588.png)

afeld commented on app/controllers/feedback_controller.rb:
```diff
@@ -0,0 +1,15 @@
+  def create
+    fields = [:bug, :context, :expected, :actually, :comments, :satisfaction, :referral]
+    fields = fields.select {|key| !params[key].blank?}
+    form_values = fields.map {|key| [key, params[key]]}
```
Not sure that it makes a *huge* difference, but this is basically what [strong parameters](http://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters) are for.
